pkgs/unified_analytics/lib/src/ga_client.dart

```diff
@@ -48,12 +48,16 @@ class GAClient {
   /// Receive the payload in Map form and parse
   /// into JSON to send to GA
   Future<http.Response> sendData(Map<String, Object?> body) {
-    return _client.post(
+    return _client
+        .post(
       Uri.parse(postUrl),
       headers: <String, String>{
         'Content-Type': 'application/json; charset=UTF-8',
       },
       body: jsonEncode(body),
-    );
+    )
+        .catchError((error) {
+      return Future<http.Response>.value(http.Response(error.toString(), 500));
```

Comment:
Is it worth noting this behaviour in the Dartdoc, just so it's clear to callers that they don't need to handle it, and a response with a 500 code isn't necessarily from the server?